transport/peer.go

```diff
@@ -0,0 +1,48 @@
+package transport
+
+//go:generate mockgen -destination=transporttest/peer.go -package=transporttest go.uber.org/yarpc/transport PeerIdentifier,Peer
+
+// PeerConnectionStatus maintains information about the Peer's connection state
+type PeerConnectionStatus int
+
+const (
+	// PeerUnavailable indicates the Peer is unavailable for requests
+	PeerUnavailable PeerConnectionStatus = iota
+
+	// PeerConnecting indicates the Peer is in the process of connecting
+	PeerConnecting
+
+	// PeerAvailable indicates the Peer is available for requests
+	PeerAvailable
```

Comment:
Why not "PeerConnected"? PeerAvailable conveys a bit more info than "connected" though. For example, you could be connected in lame duck mode. I just want to double check that "Available" was the correct semantic here.